config/options.go

```diff
@@ -475,7 +476,7 @@ type OptionsUpdater interface {
 
 // Checksum returns the checksum of the current options struct
 func (o *Options) Checksum() string {
-	hash, err := hashstructure.Hash(o, nil)
+	hash, err := hashstructure.Hash(o, &hashstructure.HashOptions{Hasher: xxhash.New()})
```

Comment:
👀 hashstructure's default hash fn is FNV which is suboptimal for a bunch of reasons. We were pulling in xxhash anyway so I figured we should use it.  Far from a promising bottleneck in both speed and quality, but it felt like a pretty "free" change. Let me know what you think 

- https://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed
- https://cyan4973.github.io/xxHash/
- https://news.ycombinator.com/item?id=10674723

Reply:
No concern as long as we can still cast the result to `float64` for the info metric.  The test in `/internal/telemetry/metrics` should probably be updated with sample data from this hash if it is significantly different output.

Reply:
They should be identical, both implement the `Hash64` interface and return a `Sum64() uint64`; I ... vaguely remember but forgot... we cast to a float and not a uint64 for some reason to have to do with prometheus right?

Reply:
Oh, actually we go uint64->float so should work as-is.

Reply:
👍